src/bit-fields-macros/src/parser.rs

```diff
@@ -0,0 +1,946 @@
+    #[error("Failed to get flag index: {0}")]
+    IndexFlag(IndexFlagError),
+    /// Unknown attribute supplied on member.
+    #[error(
+        "Unknown attribute supplied on member. Supported attributes are `#[skip]` and \
+         `#[doc=\"..\"]`."
+    )]
+    UnknownAttribute,
+    /// Rustdoc comment missing enclosing " characters.
+    #[error("Rustdoc comment missing enclosing \" characters.")]
+    RustdocUnenclosed,
+    /// Badly defined members.
+    #[error("Badly defined members: {0:?}.")]
+    BadlyDefinedMembers(Vec<String>),
+    /// Expected commma
+    #[error("Expected commma.")]
+    ExpectedComma,
+    /// Cannot define member as indexing into bit range, as bit range does not exist.
+    #[error(
+        "Cannot define member on bit range, as bit range does not exist. Bit ranges need to be \
+         defined before members which reference them."
+    )]
+    IndexingMissingRange,
+    /// Cannt define member as indexing into bit flag, as you cannot index a bit flag.
+    #[error("Cannt define member as indexing into bit flag, as you cannot index a bit flag.")]
+    IndexingBitFlag,
+    /// Malformed bit range index.
+    #[error("Malformed bit range index.")]
+    IndexingMalformed,
+    /// Members which are present in the conversion to/from sets of flags and maps of fields cannot
+    /// have overlapping bits as this leads to undefined behavior in as the value of the bit field
+    /// would depend on the ordering of keys in a `HashMap` and/or `HashSet` (which is
+    /// inconsistent). Thus this case would enable the same data to produce to bit fields of
+    /// different values.
+    #[error(
+        "Members which are present in the conversion to/from sets of flags and maps of fields \
+         cannot have overlapping bits as this leads to undefined behavior in as the value of the \
+         bit field would depend on the ordering of keys in a `HashMap` and/or `HashSet` (which is \
+         inconsistent). Thus this case would enable the same data to produce to bit fields of \
+         different values."
+    )]
+    OverlappingNonSkipped,
+}
+
+/// Advantces parser iterator expecting `base+1`th token to be `None` or a comma.
+fn advance_iter<'a>(
+    iter: &mut std::slice::Iter<'a, TokenTree>,
+    tail: &[TokenTree],
+    base: usize,
+) -> Result<(), crate::ProcError<BitFieldMembersParserIterError>> {
+    // Move past this identified slice.
+    iter.nth(base);
+    // Match comma
+    #[allow(clippy::pattern_type_mismatch)]
+    match tail.first() {
+        // If no tailing comma, we also we expect this to be end
+        None => Ok(()),
+        // If tailing comma
+        Some(TokenTree::Punct(comma)) if comma.as_char() == ',' => {
+            // Advance over comma
+            iter.next();
+            Ok(())
+        }
+        // Else if not end and next token is not comma
+        Some(token) => Err((token.span(), BitFieldMembersParserIterError::ExpectedComma)),
+    }
```

Comment:
```suggestion
    match tail.first() {
        // If no tailing comma, we also we expect this to be end
        None => Ok(()),
        // If tailing comma
        Some(&TokenTree::Punct(ref comma)) if comma.as_char() == ',' => {
            // Advance over comma
            iter.next();
            Ok(())
        }
        // Else if not end and next token is not comma
        Some(token) => Err((token.span(), BitFieldMembersParserIterError::ExpectedComma)),
    }
```

Reply:
This change is not needed due to the global `#[allow(clippy::pattern_type_mismatch)]` which exists to avoid these situations where the change would seem to make the code less readable in my opinion.